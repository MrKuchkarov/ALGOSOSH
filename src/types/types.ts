import {ElementStates} from "./element-states";

export type TCircleItem = {
    _id?: string
    item: string;
    state: ElementStates;
}

export enum SortName {
    bubble = "пузырёк",
    select = "выбор",
}

export type TArrayItem = {
    item: number;
    state: ElementStates;
}

export type TQueueItem = TCircleItem & {
    head?: string;
}

export enum position {
    head = "head",
    tail = "tail",
}

export enum ElementColors {
    Default = "#0032ff",
    Changing = "#d252e1",
    Modified = "#7fe051",
}