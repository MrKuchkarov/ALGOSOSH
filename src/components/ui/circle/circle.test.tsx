import {render} from "@testing-library/react";
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

type CircleProps = {
    letter?: string;
    head?: string | JSX.Element;
    tail?: string | JSX.Element;
    index?: number;
    isSmall?: boolean;
    state?: ElementStates;
}

describe("Testing circle component", () => {
    const testSnapshot = (props: CircleProps) => {
        const { container } = render(<Circle {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    };
    it("Should be without a letter", () => {
        testSnapshot({});
    });
    it("Should be with a letter", () => {
        testSnapshot({ letter: "a" })
    });
    it("Should be with head", () => {
        testSnapshot({ head: "7" })
    });
    it("Should be with a react element in head", () => {
        testSnapshot({ head: <Circle /> })
    });
    it("Should be with tail", () => {
        testSnapshot({ tail: "7" })
    });
    it("Should be with a react element in tail", () => {
        testSnapshot({ tail: <Circle /> });
    });

    it("Should be with index", () => {
        testSnapshot({ index: 0 });
    });

    it("Props isSmall should be true", () => {
        testSnapshot({ isSmall: true });
    });

    it("Should be default state", () => {
        testSnapshot({ state: ElementStates.Default });
    });

    it("Should be changing state", () => {
        testSnapshot({ state: ElementStates.Changing });
    });

    it("Should be modified state", () => {
        testSnapshot({ state: ElementStates.Modified });
    });
});