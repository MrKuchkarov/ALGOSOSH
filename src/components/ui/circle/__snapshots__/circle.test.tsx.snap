// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing circle component Props isSmall should be true 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be changing state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be default state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be modified state 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be with a letter 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      a
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be with a react element in head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be with a react element in tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Testing circle component Should be with head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    7
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Testing circle component Should be with index 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Testing circle component Should be with tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    7
  </div>
</div>
`;

exports[`Testing circle component Should be without a letter 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
